eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/EventualCassandra.scala

```diff
@@ -120,28 +185,35 @@ object EventualCassandra {
               statement(key, segment.nr, range).map { record => (record, segment) }
             }
 
-            read(from, Segment(from, head.segmentSize))
-              .chain { case (record, segment) =>
-                for {
-                  from    <- record.seqNr.next[Option]
-                  segment <- segment.next(from)
-                } yield {
-                  read(from, segment)
+            val events = 
+              read(from, Segment(from, head.segmentSize))
+                .chain { case (record, segment) =>
+                  for {
+                    from    <- record.seqNr.next[Option]
+                    segment <- segment.next(from)
+                  } yield {
+                    read(from, segment)
+                  }
                 }
-              }
-              .map { case (record, _) => record }
-              .stateful(from) { case (seqNr, record) =>
-                if (seqNr <= record.seqNr) {
-                  val seqNr1 = record
-                    .seqNr
-                    .next[Option]
-                  (seqNr1, Stream[F].single(record))
-                } else {
-                  val msg = s"Data integrity violated: seqNr $seqNr duplicated in multiple records from eventual journal, key $key"
-                  val err = new JournalError(msg)
-                  (seqNr.some, err.raiseError[F, EventRecord[EventualPayloadAndType]].toStream)
+                .map { case (record, _) => record }
+
+            if (dataIntegrity.seqNrUniqueness) {
+              events
+                .stateful(from) { case (seqNr, record) =>
+                  if (seqNr <= record.seqNr) {
```

Comment:
why include check for equality? shouldn't it be just `<` instead of `<=`?

Reply:
Would be better asking author of the code, but AFAIK the reason is in `seqNr` value - its _not_ last seen sequence number but rather next one expected:
```
                  if (seqNr <= record.seqNr) {
                    val seqNr1 = record
                      .seqNr
                      .next[Option]
                    (seqNr1, Stream[F].single(record))
                  }
```
In case of no gaps in sequence numbers would be enough having `seqNr == record.seqNr`, but in reality gaps are happening regularly thus KJ will accept them and delegate accepting/rejecting event to the application.

Reply:
why gaps happen?

Reply:
Again, the question is not related to the PR. Lets not block it by side discussions.

I suggest asking @theide as one having much more experience in using KJ :D